daemon/cluster/cluster.go

```diff
@@ -446,11 +446,20 @@ func (c *Cluster) UnlockSwarm(req types.UnlockRequest) error {
 
 	c.mu.RLock()
 	state := c.currentNodeState()
+	if !state.IsActiveManager() {
+		if err := c.errNoManager(state); err != errSwarmLocked {
+			c.mu.RUnlock()
+			return err
+		}
+	}
+
 	nr := c.nr
 	c.mu.RUnlock()
+
 	if nr == nil || errors.Cause(state.err) != errSwarmLocked {
```

Comment:
`nr == nil` case should be handled by the `errNoManager` check as well. I think the only case left unhandled here is if `IsActiveManager` is true.